plugin/plugins/file/service/service.go

```diff
@@ -108,6 +135,7 @@ func (fss *StreamingService) ListenBeginBlock(ctx sdk.Context, req abci.RequestB
 		if _, err = dstFile.Write(stateChange); err != nil {
 			fss.stateCache = nil
```

Comment:
`fss.stateCache` should be a chan (and called `stateChan` or `dataFeed`) - this way we won't need to use lock and open a box of potential sync errors.